spec/index.adoc

```diff
@@ -2521,6 +2540,37 @@ CanStatus
   = Running
   | Stopping (List (CallOrigin, Nat))
   | Stopped
+CanChangeOrigin
+  = FromUser {
+      user_id : UserId;
+    }
+  | FromCanister {
+      canister_id : CanisterId;
+      canister_history_index : CanHistoryIndex;
+    }
+CodeDeploymentMode
+  = Install
+  | Reinstall
+  | Upgrade
+CanChangeType
```

Comment:
Why not simply
```suggestion
CanChange
```
(it’s more than the `Type`)

Reply:
Because then it clashes with the record including timestamp and change origin which are independent of the type of the change (e.g., code deployment or controllers change). Inlining this type into `CanChange` below would also only work in the textual spec, not in the actual code as far as I can see.

Reply:
An alternative would be to rename the `CanChange` below to `CanChangeEntry`, and rename this to `CanChange`?